creator/ingest_runs/genomic_data_loader.py

```diff
@@ -0,0 +1,397 @@
+        (also referred to as "harmonized" files) in _manifest_df_.
+        Return the DataFrame used for loading, which is useful for testing.
+
+        1. Load harmonized genomic file metadata (size, filepath, etc)
+        2. Link the harmonized files to their related specimens
+        3. Link the harmonized files to their related sequencing experiments
+
+        The last two steps need to be done using the harmonized files'
+        source genomic files already registered in Data Service since the
+        source files have direct relations to the specimens and sequencing
+        experiments.
+        """
+        self.logger.info(
+            "Begin the ingest process for genomic workflow output files"
+        )
+        # -- Step 1: Load harmonized genomic files --
+        genomic_df = self.load_harmonized_genomic_files(manifest_df)
+
+        # -- Step 2: Link harmonized genomic files to specimens --
+        self.load_specimen_harmonized_gf_links(genomic_df)
+
+        # -- Step 3: Link harmonized genomic files to sequencing experiments --
+        genomic_df = self.load_seq_exp_harmonized_genomic_files(genomic_df)
+
+        # Check to make sure everything loaded correctly
+        self._validate_load(genomic_df)
+
+        return genomic_df
+
+    def load_harmonized_genomic_files(self, manifest_df):
+        """
+        Load harmonized genomic file metadata from the manifest into the
+        DataService.Scrape S3 to obtain file metadata needed.
+
+        After loading, grab the generated kf_ids from the ingest lib cache,
+        join them to _genomic_df_ and then return _genomic_df_. This DataFrame
+        is used in the rest of the ingest process
+        """
+        assert not manifest_df.empty, "Empty manifest DataFrame"
+
+        # Get S3 file metadata
+        buckets = set(
+            os.path.split(row)[0] for row in manifest_df["Filepath"]
+        )
+        self.logger.info(f"Getting file metadata from S3 {buckets} ...")
+
+        file_df = utils.scrape_s3(buckets)
+
+        # Filter out folders
+        file_df["Filename"] = file_df["Key"].map(
+            lambda x: os.path.split(x)[-1]
+        )
+        file_df = file_df[~file_df["Filename"].isnull()]
+
+        genomic_df = manifest_df.merge(
+            right=file_df, on="Filepath", how="inner"
+        )
+
+        # Prepare genomic file df for ingest
+        genomic_df = self._prep_harmonized_gf_df(genomic_df)
+
+        df = genomic_df.drop_duplicates([CONCEPT.GENOMIC_FILE.ID])
+        self.logger.info(
+            f"Loading {df.shape[0]} harmonized genomic files into "
+            f"{DATASERVICE_URL}"
+        )
+
+        # Load harmonized genomic files
+        genomic_cache = self._load_entities(GEN_FILE, df)
+        genomic_df = self._get_genomic_file_kf_ids(genomic_cache, genomic_df)
+
+        return genomic_df
+
+    def load_specimen_harmonized_gf_links(self, genomic_df):
```

Comment:
Decided to put this in its own function since the other entities had their own load functions